core/lib/tracehouse/trace-processor.js

```diff
@@ -645,18 +704,37 @@ class TraceProcessor {
           parent: evt.args.data.parent,
         });
       });
+
     const frames = [...framesById.values()];
     const frameIdToRootFrameId = this.resolveRootFrames(frames);
 
+    const inspectedTreeFrameIds = [...frameIdToRootFrameId.entries()]
+      .filter(([childFrameId, rootFrameId]) => rootFrameId === mainFrameInfo.frameId) // eslint-disable-line no-unused-vars
+      .map(([child, root]) => child); // eslint-disable-line no-unused-vars
+
+    // @ts-expect-error frameIdToRootFrameId's values were casted to string. Runtime check here to confirm.
+    if (inspectedTreeFrameIds.includes(undefined)) {
+      throw new Error('Unexpected undefined in frameIdToRootFrameId');
+    }
+
     // Filter to just events matching the main frame ID, just to make sure.
-    const frameEvents = keyEvents.filter(e => e.args.frame === mainFrameIds.frameId);
+    /** @param {LH.TraceEvent} e */
+    function associatedToMainFrame(e) {
+      const frameId = e.args?.data?.frame || e.args.frame;
```

Comment:
so if this is a thing you want done for every single access of "get me the event's frame", can this be a function instead? It's just a lot everywhere and easy to overlook when writing new code.

Reply:
`frameEvents` is kind of a lie anyways. Lots of events are from the main frame but don't have a `args.data.frame` or a `e.args.frame`, it just happened that a few types of events that we care about (e.g. CLS) do have `frames` on them, so this became a thing. It can actually be misleading the other way if you expect all the main frame's events to have a `frame` arg and be in there.

I believe we've talked about getting rid of `frameEvents`/`frameTreeEvents` in the next iteration of trace-processor now that we know about the process filtering so it's already mostly limited to events we care about, but @paulirish would remember much better.

Reply:
> I believe we've talked about getting rid of `frameEvents`/`frameTreeEvents` in the next iteration of trace-processor

Yup. I have a 2nd branch that does this.  In the two cases where we use `frameTreeEvents` we could be using a larger group of events (aka processEvents) and get the same result.